src/main/java/uk/gov/ch/repository/update/EntityDataRepository.java

```diff
@@ -0,0 +1,17 @@
+package uk.gov.ch.repository.update;
+
+import org.springframework.data.jpa.repository.Query;
+import org.springframework.data.repository.PagingAndSortingRepository;
+import uk.gov.ch.model.update.OverseasEntityData;
+
+public interface EntityDataRepository extends PagingAndSortingRepository<OverseasEntityData, String> {
+
+    @Query(value = "select "
+            + "CBD.email_address "
+            + "from CORPORATE_BODY CB "
+            + "inner join corporate_body_detail CBD on CB.corporate_body_id=CBD.corporate_body_id "
+            + "where "
+            + "CB.incorporation_number = ?" ,
```

Comment:
Do we need to do any validation on incorporation_number (SQL injection)? It needs to start with "OE" and then 6 digits and I think that is what we should restrict this api to.

Reply:
@AndreKramer7654 Now handled in Controller via regex